src/System.Private.ServiceModel/tools/test/Bridge/Bridge/Program.cs

```diff
@@ -18,10 +21,174 @@ private static void Main(string[] args)
         {
             CommandLineArguments commandLineArgs = new CommandLineArguments(args);
 
+            Console.WriteLine("Specified BridgeConfiguration is:{0}{1}",
+                                Environment.NewLine, commandLineArgs.BridgeConfiguration.ToString());
+
+            // If asked to ping (not the default), just ping and return an exit code indicating its state
+            if (commandLineArgs.Ping)
+            {
+                string pingResult = PingBridge(commandLineArgs.BridgeConfiguration.BridgeHost,
+                               commandLineArgs.BridgeConfiguration.BridgePort);
+                if (pingResult == null)
+                {
+                    Console.WriteLine("The Bridge is running.");
+                    Environment.Exit(0);
+                }
+                else
+                {
+                    Console.WriteLine("The Bridge is not running: {0}", pingResult);
+                    Environment.Exit(1);
+                }
+            }
+            else if (commandLineArgs.StopIfLocal)
+            {
+                StopBridgeIfLocal(commandLineArgs);
+            }
+            else if (commandLineArgs.Stop)
+            {
+                StopBridge(commandLineArgs);
+            }
+            else
+            {
+                // Default action is starting the Bridge
+                StartBridge(commandLineArgs);
+            }
+        }
+
+        private static string PingBridge(string host, int port)
+        {
+            string bridgeUrl = String.Format("http://{0}:{1}/Bridge", host, port);
+
+            using (HttpClient httpClient = new HttpClient())
+            {
+                Console.WriteLine("Testing Bridge at {0}", bridgeUrl);
+                try
+                {
+                    var response = httpClient.GetAsync(bridgeUrl).GetAwaiter().GetResult();
+                    if (!response.IsSuccessStatusCode)
+                    {
+                        string reason = String.Format("{0}Bridge returned unexpected status code='{1}', reason='{2}'",
+                                                    Environment.NewLine, response.StatusCode, response.ReasonPhrase);
+                        if (response.Content != null)
+                        {
+                            string contentAsString = response.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+                            if (contentAsString.Length > 1000)
+                            {
+                                contentAsString = contentAsString.Substring(0, 999) + "...";
+                            }
+                            reason = String.Format("{0}, content:{1}{2}",
```

Comment:
Status code and reason are not written to Console if there is response content. Is this intentional?